### ISSUE TITLE
您好，有关coral函数的疑问。

### ISSUE THREAD

**Issue by SCXCLY** [thuml/Transfer-Learning-Library#186]
敬的作者，您好：
        非常感谢您的伟大的工作，这方便了我们这些非专业人员快速学习有关迁移学习的知识。我在研究coral中时，有以下疑问向您请教：

1、Transfer-Learning-Library/tree/master/examples/domain_generalization/image_classification)/coral.py中，第182-192行中，

   for domain_i in range(n_domains_per_batch):
            # cls loss
            y_i, labels_i = y_all[domain_i], labels_all[domain_i]
            loss_ce += F.cross_entropy(y_i, labels_i)
            # update acc
            cls_acc += accuracy(y_i, labels_i)[0] / n_domains_per_batch
            # correlation alignment loss
            for domain_j in range(domain_i + 1, n_domains_per_batch):
                f_i = f_all[domain_i]
                f_j = f_all[domain_j]
                loss_penalty += correlation_alignment_loss(f_i, f_j)

为什么需要对每一个样本计算coral损失值，可以直接对一个step中的样本直接计算损失值吗？

2、在196行中：
   loss_penalty /= n_domains_per_batch * (n_domains_per_batch - 1) / 2
为什么计算loss_penalty需要除以n_domains_per_batch * (n_domains_per_batch - 1) / 2呢？

期待您的答疑。

**Comment by thucbx99** [comment]
CORAL是为Domain Adaptation任务设计的，你看的这部分是将CORAL应用于Domain Generalization中，常见的setting是有多个domain。这时CORAL即为在所有domain中两两计算，即有了上面的domain_i和domain_j部分代码。下面这里是为了做归一化，因为显然这样的loss被计算了组合数次，归一化后这一个loss的大小就和domain数无关了

**Comment by SCXCLY** [comment]
非常感谢~

**Comment by SCXCLY** [comment]
您好，我想请问一下在Domain Adaptation中的regression中的Regressor中返回的Feature的shape必须为(mini-match,'feature_dim')吗，可以为（mini-batch，channel，’feature_dim‘）吗？能计算JMMD以及CORAL吗？因为我的backbone的最后一层为卷积层，head也为卷积层。

**Comment by thucbx99** [comment]
这种情况可以先用pytorch的view转换到2d，计算后再转换回来，目前库里可以直接调用的JMMD和CORAL是不支持3d的
